formApp/src/App.tsx

```diff
@@ -1,10 +1,92 @@
-import { useState } from "react";
-import reactLogo from "./assets/react.svg";
-import viteLogo from "/vite.svg";
+import { z } from "zod";
 import "./App.css";
+import { FormContext } from "./components/form/FormContext";
+import { InputController } from "./components/form/InputController";
+import { SelectController } from "./components/form/SelectController";
+import { CheckboxController } from "./components/form/CheckboxController";
+import { Button } from "./components/ui/button";
+
+const animals = [
+	{
+		id: "1111111",
+		label: "cat",
+	},
+	{
+		id: "2222222",
+		label: "dog",
+	},
+	{
+		id: "3333333",
+		label: "turtle",
+	},
+	{
+		id: "4444444",
+		label: "fish",
+	},
+];
+
+const schema = z.object({
+	name: z
+		.string()
+		.min(1, { message: "Name is required" }),
+	age: z.number().min(1, {
+		message: "Age must be greater than 0",
+	}),
+	chama: z.boolean().optional().default(false),
+	animals: z
+		.enum(["cat", "dog", "turtle", "fish"])
+		.default("cat"),
+});
```

Comment:
sice tento ukol není tak uplně o tom jak máš psát zod schemata... ale pls zamysli se nad tím schmatem :D to je porno